From 466e16afa0a314fe4692ad7d534463d38ef88232 Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Sat, 10 Jan 2015 10:39:04 +0800
Subject: [PATCH 215/215] atomisp2: add default depth sensor info

Add default depth sensor master/slave information.
To make depth mode work even the sensors' information is missing on
certain platform.

Change-Id: Idaa6ca317765cafae7932a61104a8660922993f3
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-16819
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/35797
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h  | 3 +++
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c     | 8 ++++++--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
index 21edf15..0221ed4 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
@@ -171,6 +171,9 @@
 
 #define ATOMISP_DEPTH_SENSOR_STREAMON_COUNT 2
 
+#define ATOMISP_DEPTH_DEFAULT_MASTER_SENSOR 0
+#define ATOMISP_DEPTH_DEFAULT_SLAVE_SENSOR 1
+
 #define DIV_NEAREST_STEP(n, d, step) \
 	round_down((2 * (n) + (d) * (step))/(2 * (d)), (step))
 
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index acce26a..412a9d4 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -1578,8 +1578,12 @@ int atomisp_stream_on_master_slave_sensor(struct atomisp_device *isp, bool isp_t
 	}
 
 	if (master == -1 || slave == -1) {
-		dev_err(isp->dev, "depth mode needs 2 sensors to be selected.\n");
-		return -EINVAL;
+		master = ATOMISP_DEPTH_DEFAULT_MASTER_SENSOR;
+		slave = ATOMISP_DEPTH_DEFAULT_SLAVE_SENSOR;
+		dev_warn(isp->dev,
+			 "depth mode use default master=%s.slave=%s.\n",
+			 isp->inputs[master].camera->name,
+			 isp->inputs[slave].camera->name);
 	}
 
 	if (isp_timeout) {
-- 
1.9.1

